src/requirements/VisitLink/VisitLinkForm.tsx

```diff
@@ -57,11 +57,7 @@ const VisitLinkForm = ({ baseFieldPath }: RequirementFormProps) => {
         placeholder="https://guild.xyz"
         onChange={(e) => {
           const position = e.target.selectionStart
-          onChange(
-            e.target.value.toLowerCase().includes("youtube")
-              ? e.target.value
-              : e.target.value.toLowerCase()
-          )
+          onChange(new URL(e.target.value).href)
```

Comment:
Lovely solution, but this will cause an error:
<img width="645" alt="image" src="https://github.com/guildxyz/guild.xyz/assets/92519134/84fc6bfe-7894-440a-b83d-dd9f27b61e8e">

Maybe you could just put it in a try-catch like this?

```ts
try {
  const newValue = new URL(e.target.value).href
  onChange(newValue)
} catch {
  onChange(e.target.value)
}
```